sql/executor.go

```diff
@@ -373,7 +373,9 @@ func (e *Executor) execRequest(ctx context.Context, session *Session, sql string
 			txnState.reset(ctx, e, session)
 			txnState.State = Open
 			txnState.autoRetry = true
-			execOpt.MinInitialTimestamp = e.ctx.Clock.Now()
+			now := e.ctx.Clock.Now()
+			txnState.sqlTimestamp = now
```

Comment:
Sorry for being late to the party, but now you're (again) taking a timestamp off the HLC for the statement timestamp, which means it's going to behave just like the actual transaction timestamp (including not changing for up to `MaxOffset`). It was my understanding that we wanted a timestamp of the local *physical* clock, i.e. `e.ctx.Clock.Physical{Now,Time}()` (which would not be a HLC timestamp, but only a wall time).